test/models/filter_test.rb

```diff
@@ -6,4 +6,19 @@ class FilterTest < ActiveSupport::TestCase
     assert Filter
   end
 
+  test "it filters developers by tags and order" do
+    tag_1, tag_2, tag_3 = create_list(:tag,3)
+    dev_1 = create(:developer, tags: [tag_1, tag_2, tag_3])
+    dev_2 = create(:developer, tags: [tag_1, tag_2])
+    dev_3 = create(:developer, tags: [tag_2, tag_3])
+
+    params = {}
+    params['Sort By'] = "Rate: Descending"
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.